ci-bench/src/cachegrind.rs

```diff
@@ -291,3 +291,29 @@ pub fn diff(baseline: &Path, candidate: &Path, scenario: &str) -> anyhow::Result
 
     Ok(diff)
 }
+
+pub(crate) fn reset_instruction_count() {
+    #[cfg(target_os = "linux")]
+    crabgrind::callgrind::zero_stats();
```

Comment:
sadly this actually does jack, because we're running this under _cachegrind_ and not _callgrind_  -- the version of valgrind we have in ci does not have a programmatic control interface for cachegrind (though does in later versions)

will have a look to see if we can use callgrind instead (we turn off the cache simulation, so they are should be very similar)

Reply:
Can we upgrade our version of valgrind?

Reply:
Unfortunately the test runner is on debian stable, installing a newer valgrind from debian unstable would work, but would give us an odd mix of the two. I think if I get into changing the ansible scripts for this, I would rather spend the time installing docker to make it easier to (eg) rev the rust compiler too.

Reply:
Will benchmarking from Docker introduce any significant overhead?

Reply:
I think some networking/filesystem performance can be worse, but pure CPU should be identical.

Reply:
callgrind seems to work well now, though renaming a bunch of types/files called CachegrindSomething remains to do.